hooks/useUpdatePoolInstances/index.ts

```diff
@@ -125,8 +127,8 @@ export const useUpdatePoolInstances = (): void => {
     }, [provider, poolsInitialized]);
 
     // update token balances and approvals when address changes
-    useEffect(() => {
-        if (provider && account && poolsInitialized) {
+    useMemo(() => {
```

Comment:
why switch to useMemo here? my understanding is useMemo is used like useCallback but for other types of variables

Reply:
should be useEffect